internal/host/hostcommands/domain_name_resolution_cmd.go

```diff
@@ -0,0 +1,64 @@
+package hostcommands
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+type domainNameResolutionCmd struct {
+	baseCmd
+	domainNameResolutionImage string
+	db                        *gorm.DB
+}
+
+func NewdomainNameResolutionCmd(log logrus.FieldLogger, domainNameResolutionImage string, db *gorm.DB) *domainNameResolutionCmd {
+	return &domainNameResolutionCmd{
+		baseCmd:                   baseCmd{log: log},
+		domainNameResolutionImage: domainNameResolutionImage,
+		db:                        db,
+	}
+}
+
+func (f *domainNameResolutionCmd) prepareParam(host *models.Host, cluster *common.Cluster) (string, error) {
+	domainName := cluster.Cluster.APIVipDNSName
```

Comment:
We should check that the domain name is not empty.  If it is, then we should not send this step (return nil instead to GetSteps).

Reply:
Done